### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by przwojwwp**
[DEMO LINK](https://przwojwwp.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,50 @@
...
+    handleSearchChange,
+  } = useTodoFilter(todos);
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      const todosData = await getTodos();
+
+      setTodos(todosData);
+      setIsTodoLoading(false);
+    };
```
Comment (natalia-klonowska) [inline src/App.tsx:33]: you should be able to catch errors if fetch fails

**Review by natalia-klonowska — CHANGES_REQUESTED** [review]
Great work 🔥 just add error handling when fetching data because now you have infinite loader ;)

File: src/components/TodoModal/useTodoModal.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      if (todoId !== null) {
+        const userData = await getUser(todoId);
+
+        setUser(userData);
+        setIsLoading(false);
```
Comment (natalia-klonowska) [inline src/components/TodoModal/useTodoModal.ts:15]: here you also need error handling ;) remember to always handle failures in your asynchronous code to ensure a better user experience and prevent uncaught errors

**Review by natalia-klonowska — CHANGES_REQUESTED** [review]

**Review by BogdanMaliuta — COMMENTED** [review]
Nice!

**Review by BogdanMaliuta — APPROVED** [review]

**Pull request closed without merging**